contrib/src/main/java/com/datatorrent/contrib/parser/FixedWidthParser.java

```
@@ -0,0 +1,489 @@
+      }
+      errorTupleCount++;
+      return;
+    }
+    try {
+      String[] values = univocityParser.parseLine(incomingString);
+      HashMap<String, Object> toEmit = new HashMap();
+      Object pojo = validateAndSet(values, toEmit);
+      if (parsedOutput.isConnected()) {
+        parsedOutput.emit(toEmit);
+        parsedOutputCount++;
+      }
+      if (out.isConnected() && clazz != null) {
+        out.emit(pojo);
+        emittedObjectCount++;
+      }
+    } catch (Exception e) {
+      if (err.isConnected()) {
+        err.emit(new KeyValPair<>(incomingString, e.getMessage()));
+      }
+      errorTupleCount++;
+      logger.error("Tuple could not be parsed. Reason {}", e.getMessage());
+    }
+  }
+
+  @Override
+  public KeyValPair<String, String> processErrorTuple(byte[] input)
+  {
+    throw new UnsupportedOperationException("Not supported");
+  }
+
+  @Override
+  public Object convert(byte[] tuple)
+  {
+    throw new UnsupportedOperationException("Not supported");
+  }
+
+  @Override
+  public void setup(Context.OperatorContext context)
+  {
+    try {
+      schema = new FixedWidthSchema(jsonSchema);
+      recordLength = 0;
+      List<FixedWidthSchema.FixedWidthField> fields = schema.getFields();
+      for (int i = 0; i < fields.size(); i++) {
+        recordLength += fields.get(i).getFieldLength();
+      }
+      createUnivocityParser();
+    } catch (Exception e) {
+      logger.error("Cannot setup Parser Reason {}", e.getMessage());
+    }
+  }
+
+  /**
+   * Activate the Parser
+   */
+  @Override
+  public void activate(Context context)
+  {
+    try {
+      if (clazz != null) {
+        setters = new ArrayList<>();
+        List<String> fieldNames = schema.getFieldNames();
+        if (fieldNames != null) {
+          for (String fieldName : fieldNames) {
+            addSetter(fieldName);
+          }
+        }
+      }
+    } catch (Exception e) {
+      logger.error("Cannot activate Parser Reason {}", e.getMessage());
```

Don't swallow exception.

---

Done.